apps/system/test/marionette/browser_context_menu_test.js

```diff
@@ -39,11 +39,36 @@ marionette('Browser Context Menu', function() {
     rocketbar.goToURL(url);
     system.shareLink();
 
-
     system.getActivityOptionMatching('sms').tap();
     var smsFrame = messagesApp.waitForAppToAppear();
     client.switchToFrame(smsFrame);
 
     assert.equal(messagesApp.Composer.messageInput.text(), url);
   });
+
+  test('save image', function() {
+    var url = server.url('sample_image.html');
+    rocketbar.goToURL(url);
+    system.gotoBrowser(url);
+
+    var image = client.helper.waitForElement('img');
+    actions.longPress(image, 0.5).perform(function() {
+      system.saveImage();
+
+      client.switchToFrame();
+      client.waitFor(function() {
```

Comment:
nit: no need to do this wait for `system.toaster.display()` will internally do a `waitForElement`.